packages/middleware-user-agent/src/user-agent-middleware.ts

```diff
@@ -36,15 +36,22 @@ export const userAgentMiddleware = (options: UserAgentResolvedConfig) => <Output
   const userAgent = context?.userAgent?.map(escapeUserAgent) || [];
   const defaultUserAgent = (await options.defaultUserAgentProvider()).map(escapeUserAgent);
   const customUserAgent = options?.customUserAgent?.map(escapeUserAgent) || [];
+
   // Set value to AWS-specific user agent header
-  headers[X_AMZ_USER_AGENT] = [...defaultUserAgent, ...userAgent, ...customUserAgent].join(SPACE);
+  const sdkUserAgentValue = [...defaultUserAgent, ...userAgent, ...customUserAgent].join(SPACE);
   // Get value to be sent with non-AWS-specific user agent header.
   const normalUAValue = [
     ...defaultUserAgent.filter((section) => section.startsWith("aws-sdk-")),
     ...customUserAgent,
```

Comment:
Now that `normalUAValue` is supplied to `x-amz-user-agent` header,  we don't need to repeat the custom user agent in both `x-amz-user-agent` and `user-agent` header. This can be removed. 
The was added here previously because `normalUAValue` is supplied to `user-agent` header. And when customer set the custom user agent, they expect it to exist in the `user-agent` header

Reply:
As per offline discussion, we're retaining `customUserAgent` as this commit is going to be reverted in future when services start logging `x-amz-user-agent`.